### ISSUE TITLE
Better explanation of yaml configuration

### ISSUE THREAD

**Issue by arubertoson** [itchyny/miv#8]
I just came across miv, I read a reddit post from 2 years ago that you're not interested in making it popular, so I guess it's mostly written by you, for you. I still believe that it would be nice to have some explanation how to write the yaml. Some options I can easily pick up on as they remind me of other managers with lazy evaluation but others are harder, mostly thinking about mapleader, mapping, insert and script.

It might be an effort, but making it a bit more accessible wouldn't hurt!

**Comment by arubertoson** [comment]
So looking a bit at the source, can't really read haskell that well but from
the generated vim files this is what I've gathered:

**name**: Alias that miv will use internally (?)
**filetype**: Will source when entering a filetype
**command**: Will source when command is run
**function**: Will source when a function is called
**mappings**: Will source when mappings are pressed (? quite unsure)
**mapmodes**: What modes the mappings should be bound to (?)
**insert**: ??
**enable**: enable if expression is true
**sync**: ??
**mapleader**: I'm guessing that when binging with <leader> the map is translated
  \<leader> = g:mapleader, when changing mapleader value the next map will be
  bound to the new value. (?)
**script**: is placed in the plugin/miv.vim file, running on vim startup
**after**: run after plugin is sourced
**before**: run just before plugin is sourced
**dependon**: Depends on another plugin, (will source when other plugin is loaded?)
**dependby**: ??
**loadafter**: Load after given plugin
**loadbefore**: Load before given plugin
**submodule**: ??

Would really appriciate further clarifications!
